### PR TITLE
[WIP] Adding EdgeSource

### PR DESCRIPTION
**Pull request by n3wscott**
Fixes https://github.com/knative/eventing-sources/issues/33

## Proposed Changes

  * Add EdgeSource 

**Release Note**
```release-note
NONE
```

### CODE REVIEW

File: config/crds/sources_v1alpha1_edgesource.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+metadata:
+  creationTimestamp: null
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: edgesources.sources.eventing.knative.dev
+spec:
+  group: sources.eventing.knative.dev
+  names:
+    kind: EdgeSource
+    plural: edgesources
```
Comment (matzew) [inline config/crds/sources_v1alpha1_edgesource.yaml]: `short: edge` ?

Comment (n3wscott) [reply config/crds/sources_v1alpha1_edgesource.yaml]: I think short names should be added very carefully.

File: config/samples/sources_v1alpha1_edgesource.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: sources.eventing.knative.dev/v1alpha1
+kind: EdgeSource
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: edgesource-sample
+spec:
+  # Add fields here
```
Comment (vaikas-google) [inline config/samples/sources_v1alpha1_edgesource.yaml]: Could we add a sample here that actually shows a sample? :)

File: pkg/controller/edgesource/resources/job.go [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+	// source wrapper expects it to be valid json.
+	if len(marshaledContext) == 0 {
+		marshaledContext, err = json.Marshal(JobContext{})
+		if err != nil {
+			return nil, err
+		}
+	}
+	//encodedContext := base64.StdEncoding.EncodeToString(marshaledContext)
+
+	//marshalledTrigger, err := json.Marshal(trigger)
```
Comment (vaikas-google) [inline pkg/controller/edgesource/resources/job.go]: Remove these?

File: pkg/controller/edgesource/resources/job.go [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+					Env: []corev1.EnvVar{
+						{
+							Name:  string(EnvVarOperation),
+							Value: string(op),
+						},
+						{
+							Name:  string(EnvVarSink),
+							Value: source.Status.SinkURI,
+						},
+						//{
```
Comment (vaikas-google) [inline pkg/controller/edgesource/resources/job.go]: and these?
Or is there a TODO here or ??

**Comment by n3wscott** [comment]
/hold

@evankanderson is going to take over this. via fork or dealers choice.

**Comment by n3wscott** [comment]
@evankanderson what was the end result? Did this happen or should I pick this back up?

**Pull request closed without merging**